JS/Lesson16/gulcin/script.js

```diff
@@ -173,16 +200,153 @@ isPositiveNegativeZero(0);
 // 25.0 – 29.9	Overweight
 // 30.0 and Above	Obesity
 
+function findBodyMassIndex(height, weight) {
+	const bodyMassIndex = weight / (height * height);
+	if (bodyMassIndex > 0 && bodyMassIndex < 18.5) {
+		console.log(`Your height is ${height} meters, your weight is ${weight} kilograms. Your BMI: ${bodyMassIndex} You are underweight.`);
+	} else if (bodyMassIndex >= 18.5 && bodyMassIndex <= 24.9) {
+		console.log(`Your height is ${height} meters, your weight is ${weight} kilograms. Your BMI: ${bodyMassIndex} You are healthy.`);
+	} else if (bodyMassIndex > 24.9 && bodyMassIndex <= 29.9) {
+		console.log(`Your height is ${height} meters, your weight is ${weight} kilograms. Your BMI: ${bodyMassIndex} You are overweight.`);
+	} else if (bodyMassIndex > 29.9) {
+		console.log(`Your height is ${height} meters, your weight is ${weight} kilograms. Your BMI: ${bodyMassIndex} You may have obesity.`);
+	} else {
+		console.log(`Not valid result.`);
+	}
+}
+
+findBodyMassIndex(1.7, 50);
+findBodyMassIndex(1.7, 60);
+findBodyMassIndex(1.75, 85);
+findBodyMassIndex(1.6, 105);
+findBodyMassIndex('knvfv', -65);
+
 // 9. Write a program that takes in a person's age and checks if they are a child (age 0-12), teenager (age 13-19), adult (age 20-64), or senior citizen (age 65 and above) using if else statement.
 
+function findAge(age) {
+  if (age > 0 && age <= 12) {
+	console.log(`Your age is ${age}. You are a child.`);
+  } else if (age > 12 && age <= 19) {
+	console.log(`Your age is ${age}. You are a teenager.`);
+  } else if (age > 19 && age <= 64) {
+	console.log(`Your age is ${age}. You are an adult.`);
+  } else if (age > 65) {
+    console.log(`Your age is ${age}. You are a senior citizen.`)
+  } else {
+	console.log('Enter a valid age number.');
+  }
+}
+
+findAge(8);
+findAge(13);
+findAge(33);
+findAge(75);
+findAge(-45);
+
 // 10. Write a program that takes age as a number and whether they are a student (yes or no). If the age is less than 18 and the user is a student, print "You are a student and not old enough to vote". If the age is between 18 and 65 (inclusive) and the user is not a student, print "You are eligible to vote". If the age is greater than 65, print "You are a senior citizen". If the user is a student but their age is greater than or equal to 18, print "You are old enough to vote".
 
+function checkToVote(age, student = 'no') {  // I provided a default value for student parameter just for senior citizen condition, because we will pass only age argument.
```

Comment:
Whenever you need to make a decision like this in programming where an answer is yes or no, use a boolean type for the value. It will make your code a lot less error-prone. Also, well done adding the default value for the function parameter!

Reply:
Then I should say true or false, right? like student === true or student === false